### PR TITLE
XW-1700 | Add dropdowns to Global Navigation

### PR DESCRIPTION
**Pull request by rogatty**
## Links

* https://wikia-inc.atlassian.net/browse/XW-1700
* http://emberjs.com/blog/2016/01/15/ember-2-3-released.html#toc_contextual-components
* http://emberjs.com/api/classes/Ember.Templates.helpers.html#method_component

## Description

Use an idea from [ember-basic-dropdown](https://github.com/cibernox/ember-basic-dropdown/blob/master/addon/templates/components/basic-dropdown.hbs) to create `wds-dropdown` component which is reusable and potentially highly customizable.

The simplest way to use it is:
```
{{#design-system/wds-dropdown as |dropdown|}}
	{{#dropdown.toggle}}Toggle{{/dropdown.toggle}}
	{{#dropdown.content}}Content{{/dropdown.content}}
{{/design-system/wds-dropdown}}
```

## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: front/main/app/templates/components/design-system/global-navigation-dropdown.hbs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		{{/if}}
+	{{/dropdown.toggle}}
+	{{#dropdown.content
+		class='wds-global-navigation__dropdown-content'
+		dropdownRightAligned=dropdownRightAligned
+	}}
+		<ul class="wds-list wds-is-linked">
+			{{#each model.links as |link|}}
+				<li>
+					{{#if (equal link.type 'link-authentication')}}
```
Comment (hakubo) [inline front/main/app/templates/components/design-system/global-navigation-dropdown.hbs]: this looks heavy - any idea how to optimize it?
{{component 'component-name' model=link}} maybe?

Comment (rogatty) [inline front/main/app/templates/components/design-system/global-navigation-dropdown.hbs]: Great idea, fixed in https://github.com/Wikia/mercury/pull/2785/commits/42afb7c7b38950f28c93ec266789063289c73efd

**Comment by kvas-damian** [comment]
LGTM

File: front/main/tests/integration/components/design-system/wds-dropdown-test.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+test('yields toggle and content', function (assert) {
+	this.render(hbs`
+		{{#design-system.wds-dropdown as |dropdown|}}
+			{{#dropdown.toggle}}Toggle{{/dropdown.toggle}}
+			{{#dropdown.content}}Content{{/dropdown.content}}
+		{{/design-system.wds-dropdown}}
+	`);
+
+	assert.ok(
+		new RegExp(`Toggle\n${toggleSvgPattern}`).test(
```
Comment (tomasznapieralski) [inline front/main/tests/integration/components/design-system/wds-dropdown-test.js]: can't we just check if inside `wds-dropdown` there is element with class `wds-dropdown__toggle`?

**Comment by tomasznapieralski** [comment]
1 question, LGTM

**Pull request merged**